### PR TITLE
NAS-101624 / 11.3 / Fix bad usage of plugin names

### PR DESCRIPTION
**Pull request by skarekrow**
These are artifacts of a time when iocage didn’t support multiple plugins at the same time.

This PR should fix the following behaviors:
- Update
- Upgrade
- Destruction on Exception

NAS-101624

Signed-off-by: Brandon Schneider <github@bschneider.email>

### CODE REVIEW

**Review by sonicaj — COMMENTED** [review]
We should cover https://github.com/iocage/iocage/blob/ce967e2e0caded851ff2300fc1ed5b3b6a15f503/iocage_lib/ioc_plugin.py#L779 this usage as well to reflect `fetch_plugin` changes

**Comment by skarekrow** [comment]
> We should cover
> 
> https://github.com/iocage/iocage/blob/ce967e2e0caded851ff2300fc1ed5b3b6a15f503/iocage_lib/ioc_plugin.py#L779
> this usage as well to reflect `fetch_plugin` changes

What do you mean? That shouldn't require any changes.

**Comment by sonicaj** [comment]
> > We should cover
> > https://github.com/iocage/iocage/blob/ce967e2e0caded851ff2300fc1ed5b3b6a15f503/iocage_lib/ioc_plugin.py#L779
> > 
> > this usage as well to reflect `fetch_plugin` changes
> 
> What do you mean? That shouldn't require any changes.

We only allow 3 parameters now instead of four ;) So this should be updated to reflect that

File: iocage_lib/ioc_plugin.py [hunk 1]
```diff
@@ -149,18 +148,18 @@ def fetch_plugin(self, _json, props, num, accept_license):
...
-                                                   pkg, props, repo_dir)
-            self.__fetch_plugin_post_install__(conf, _conf, jaildir, jail_name)
-        except Exception as e:
+            jaildir, _conf, repo_dir = self.__fetch_plugin_create__(props)
+            self.__fetch_plugin_install_packages__(
+                jaildir, conf, pkg, props, repo_dir
+            )
+            self.__fetch_plugin_post_install__(conf, _conf, jaildir)
+        except BaseException as e:
+            self.keep_jail_on_failure = False
```
Comment (sonicaj) [inline iocage_lib/ioc_plugin.py]: I am sorry i am not sure if i follow this 🤔 Why do we hardcode this value here on an exception ?
Do we want to destroy plugins always on failure ?

**Review by sonicaj — COMMENTED** [review]
Looks good to me ;) Just one minor concern

File: iocage_lib/ioc_plugin.py (hunk 1 above)
Comment (skarekrow) [reply iocage_lib/ioc_plugin.py]: Debug :D I can't believe I forgot it.

Comment (sonicaj) [reply iocage_lib/ioc_plugin.py]: Happens to the best of us ;)

**Review by sonicaj — APPROVED** [review]
Thank you

**Pull request merged**